src/main/java/software/amazon/qldb/QldbSession.java

```
@@ -128,6 +128,8 @@ boolean isClosed() {
                 if (executionContext.retryAttempts() >= retryPolicy.maxRetries()) {
                     throw sce;
                 }
+            } catch (Exception e) {
+                noThrowAbort(transaction);
```

You need to re-throw the exception here after abort.

---

Good catch - fixed in update